api/src/utils/transaction.ts

```diff
@@ -0,0 +1,15 @@
+import type { Knex } from 'knex';
+
+/**
+ * Execute the given callback in a transaction if the current knex state isn't a transaction yet.
+ *
+ * Can be used to ensure the nested callback is run within a transaction, without nesting
+ * transactions
+ */
+export const transaction = async <T = unknown>(knex: Knex, callback: (knex: Knex) => Promise<T>): Promise<T> => {
+	if (knex.isTransaction) {
+		return await callback(knex);
+	} else {
+		return await knex.transaction(async (trx) => await callback(trx));
+	}
+};
```

Comment:
Should we not directly return the promises here to save the overhead of creating a couple new promise objects which are immediately awaited/collapsed. Theoretically it should make a difference at scale but going to be hard to prove with a test 🤔 
```suggestion
export const transaction = <T = unknown>(knex: Knex, callback: (knex: Knex) => Promise<T>): Promise<T> => {
	if (knex.isTransaction) {
		return callback(knex);
	} else {
		return knex.transaction((trx) => callback(trx));
	}
};
```